### PR TITLE
[Iwara] Add new metadata extractors

### PR DESCRIPTION
**Pull request by AlexAplin**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Adds upload_date, uploader, description, comment_count, like_count, and view_count metadata extraction. New pull request opened for changes requested in #13779 as the original request referred to a deleted repo.

### CODE REVIEW

File: youtube_dl/extractor/iwara.py [hunk 1]
```diff
@@ -71,6 +82,23 @@ def _real_extract(self, url):
...
+
+        description = clean_html(get_element_by_class('field-type-text-with-summary', webpage))
+
+        comments_id = get_element_by_id('comments', webpage)
+        comment_count = int_or_none(self._html_search_regex(
+            r'([\d,]+)', get_element_by_class('title', comments_id), 'comment_count', fatal=False))
+
+        node_views_class = get_element_by_class('node-views', webpage)
+        like_count = str_to_int(self._html_search_regex(
+            r'glyphicon-heart[^>]+><\/i>\s+([\d,]+)', node_views_class, 'like_count', fatal=False))
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Forward slash does not need escaping.

Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Breaks if `node_views_class` is `None`.

File: youtube_dl/extractor/iwara.py [hunk 2]
```diff
@@ -71,6 +82,23 @@ def _real_extract(self, url):
...
+        view_count = str_to_int(self._html_search_regex(
+            r'glyphicon-eye-open[^>]+><\/i>\s+([\d,]+)', node_views_class, 'view_count', fatal=False))
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Breaks if `node_views_class` is `None`.

File: youtube_dl/extractor/iwara.py [hunk 3]
```diff
@@ -71,6 +82,23 @@ def _real_extract(self, url):
         title = remove_end(self._html_search_regex(
             r'<title>([^<]+)</title>', webpage, 'title'), ' | Iwara')
 
+        upload_date = unified_strdate(self._html_search_regex(
+            r'作成日:(\d{4}-\d{2}-\d{2})', webpage, 'upload_date', fatal=False))
+
+        uploader = get_element_by_class('username', webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Class `username` matches multiple elements.

File: youtube_dl/extractor/iwara.py [hunk 4]
```diff
@@ -71,6 +82,23 @@ def _real_extract(self, url):
...
+
+        description = clean_html(get_element_by_class('field-type-text-with-summary', webpage))
+
+        comments_id = get_element_by_id('comments', webpage)
+        comment_count = int_or_none(self._html_search_regex(
+            r'([\d,]+)', get_element_by_class('title', comments_id), 'comment_count', fatal=False))
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Breaks on `None` `comments_id`, breaks on `None` `title`.

**Comment by AlexAplin** [comment]
@dstftw Please review at your leisure.

**Comment by Rektis** [comment]
With the recent iwara update a week or so ago, downloading from iwara with youtube-dl was broken. This was fixed in https://github.com/rg3/youtube-dl/commit/edb0e17188a1197afae1a0d594e4aab7d27bbcf2 (thank you, dstftw), but updating to that made meta-data extraction broken for me.

Using AlexAplin's extractor with the fix above pasted in made everything work perfectly again.

Not being a coder, I can't attest to how well it's actually made, but I can attest to it making everything working again, at least. Just like last time he fixed the extractor. 

Thank you so much, AlexAplin!

**Comment by Rektis** [comment]
With the recent iwara update a week or so ago, downloading from iwara with youtube-dl was broken. This was fixed in https://github.com/rg3/youtube-dl/commit/edb0e17188a1197afae1a0d594e4aab7d27bbcf2 (thank you, dstftw), but updating to that made meta-data extraction broken for me.

Using AlexAplin's extractor with the fix above pasted in made everything work perfectly again.

Not being a coder, I can't attest to how well it's actually made, and it seems that metadata extraction doesn't work for public but unlisted videos (like [this ](https://ecchi.iwara.tv/videos/k0we2hqkrwszrb1bp?language=en)one), but I can attest to it making everything working for listed videos again, at least. Just like last time he fixed the extractor. 

Thank you so much, AlexAplin!

**Comment by AlexAplin** [comment]
I see no problems with metadata on that specific video.

**Comment by Rektis** [comment]
Interesting that that specific video works for you. I'll post a log of an example video that doesn't work for me, though I cut out other links from the batch file, because it doesn't seem relevant. 

I'm not sure whether this one is a listed (= normal videos you find by browsing) or unlisted (= public and not set to private, but can only be found by knowing the link) one, and whether this really matters. 
But this one doesn't work for me either way.

(Maybe I should post this on the issue tracker instead, though.)


---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.07.21*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [X] I've **verified** and **I assure** that I'm running youtube-dl **2018.07.21**

### Before submitting an *issue* make sure you have:
- [X] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [X] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [X] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [X] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### Full verbose output:


```

C:\Users\USERNAME>py -m youtube_dl -o "F:/HT/HT/VID/youtube-dl_automatic/iwara/%(uploader)s/%(upload_date)s --- %(id)s --- %(uploader)s --- %(title)s.%(ext)s" --batch-file="F:/HT/HT/VID/youtube-dl_automatic/dl_list.txt" --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-o', u'F:/HT/HT/VID/youtube-dl_automatic/iwara/%(uploader)s/%(upload_date)s --- %(id)s --- %(uploader)s --- %(title)s.%(ext)s', u'--batch-file=F:/HT/HT/VID/youtube-dl_automatic/dl_list.txt', u'--verbose']
[debug] Batch file urls: [u'https://ecchi.iwara.tv/videos/ej2xwf7nzqckvp1gg?language=en']
[debug] Encodings: locale cp932, fs mbcs, out cp932, pref cp932
[debug] youtube-dl version 2018.07.21
[debug] Python version 2.7.14 (CPython) - Windows-10-10.0.15063
[debug] exe versions: ffmpeg N-90710-gaedbf1640c, ffprobe N-90710-gaedbf1640c
[debug] Proxy map: {}
[Iwara] ej2xwf7nzqckvp1gg: Downloading webpage
[Iwara] ej2xwf7nzqckvp1gg: Downloading JSON metadata
WARNING: unable to extract upload_date; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
WARNING: unable to extract uploader; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on u'https://ruko.iwara.tv/file.php?expire=1532988348&hash=646f188d50aff30d3b112b14cd970f4a0699cf72&file=2018%2F07%2F17%2F1531830792_ej2Xwf7NzQcKvp1gG_Source.mp4&op=dl&r=0'
[download] Destination: F:\HT\HT\VID\youtube-dl_automatic\iwara\NA\NA --- ej2xwf7nzqckvp1gg --- NA --- 【r18MMD】回レ!雪月花【つみミクさん】.mp4
[download] 100% of 120.84MiB in 01:20


...
<end of log>
```

---

### Example URLs:
- Single video: https://ecchi.iwara.tv/videos/ej2xwf7nzqckvp1gg?language=en

---

### Description of your *issue*, suggested solution and other information

The video downloads fine, but some of the video's metadata isn't able to be downloaded, specifically `upload_date` and `uploader`. I'd like these to download, too.
The file ID and filename seem to download.

**Comment by AlexAplin** [comment]
Third video is a problem with the way the comment count is extracted, easy enough to fix. Language causes metadata to change because the extraction depends on the language. That can be fixed by forcing the webpage retrieval to be JP. I'll fix it eventually.

**Comment by Rektis** [comment]
Any updates?

**Pull request closed without merging**